### PR TITLE
Skills - Oath

### PR DESCRIPTION
**Pull request by JesterWizard**
All 8 variants of the Oath skills. 

+4 to the respective stat at the start of the turn if an adjacent ally is present (and removed the following turn if not).

I had to expand `UNIT_STAT_DEBUFF_MAX` in `debuff.h` from 32 to 64 to accomodate the buffs for the Oath skills.

### CODE REVIEW

File: Data/SkillSys/SkillInfo.c [hunk 1]
```diff
@@ -2243,4 +2243,68 @@ const struct SkillInfo gSkillInfos[MAX_SKILL_NUM + 1] = {
...
+        .icon = GFX_SkillIcon_OathDefense,
+      },
+#endif
+
+#if (defined(SID_OathLuck) && COMMON_SKILL_VALID(SID_OathLuck))
+    [SID_OathLuck] = {
+        .name = MSG_SKILL_OathLuck_NAME,
+        .desc = MSG_SKILL_OathLuck,
+        .icon = GFX_SkillIcon_OathLuck,
+      },
```
Comment (MokhaLeee) [inline Data/SkillSys/SkillInfo.c:2260]: Indentation issues

**Comment by MokhaLeee** [comment]
1. Solve the indentation issues
2. Solve the merge conflicts

**Comment by JesterWizard** [comment]
> 1. Solve the indentation issues
> 
>     2. Solve the merge conflicts

Done

**Comment by MokhaLeee** [comment]
In fact, there is a serious bug in this PR: StatDebuff index cannot expand to 32 bacause it is a bitmask of u32 and need allocate memory in SRAM. There is also a problem with the gameplay design, as with abilities in Heros, each team's buff should be triggered at the start of their respective turn. But these skill made sense though, and I decided to fix it myself

**Pull request merged**